### PR TITLE
Fix VLOG?_DEFAULT macros usability.

### PR DESCRIPTION
**Pull request by cloudhan**
**Description**: Describe your changes.

Currently `VLOGS_DEFAULT` and `VLOGF_DEFAULT` macros results no log in user code due to internal bug.

https://github.com/microsoft/onnxruntime/blob/7443edb0bfc0e147b660b8f43e0e73df8117a98a/onnxruntime/core/common/logging/logging.cc#L90
`default_max_vlog_level` has a default value at https://github.com/microsoft/onnxruntime/blob/7443edb0bfc0e147b660b8f43e0e73df8117a98a/include/onnxruntime/core/common/logging/logging.h#L107-L110

but no `LoggingManager` is created with that param, so ORT always have vlog level set to `-1`, that is effectly disabled, so it is changed to populate the value from environment variable.

Secondly, the `max_vlog_level_` in `logging.h` is changed to runtime check. Originally, is is checked on Logger creation time, however, the severity can be changed at runtime. This cause a bug. Also, the pybind create env with `kWARNING` by default. https://github.com/microsoft/onnxruntime/blob/7443edb0bfc0e147b660b8f43e0e73df8117a98a/onnxruntime/python/onnxruntime_pybind_state.cc#L1554-L1558
If we then change the log level via `ort.set_default_logger_severity(log_level)`, the `VLOG?_DEFAULT` will not be enabled again.

### CODE REVIEW

**Comment by cloudhan** [comment]
@skottmckay Could you please take a look? This is also cherrypick-ed from opencl poc. I might need to improve it a bit.

**Comment by cloudhan** [comment]
Please note that VLOG is nop if in release mode. So this only useful to developer.

**Comment by skottmckay** [comment]
Can you set the severity to VERBOSE and the verbosity level to the VLOG level you want in the session options prior to creating the inference session as those are used in the call to create the session logger [here](https://github.com/microsoft/onnxruntime/blob/ccd7a2d8407c37c264258cb79725cd2cb6b31328/onnxruntime/core/session/inference_session.cc#L2270-L2271)?

https://onnxruntime.ai/docs/api/c/struct_ort_api.html
```
OrtStatus * | SetSessionLogVerbosityLevel (OrtSessionOptions *options, int session_log_verbosity_level)
OrtStatus * | SetSessionLogSeverityLevel (OrtSessionOptions *options, int session_log_severity_level)
```

Or are you trying to log via the default logger (which generally we try to avoid if the session logger is available)?

**Comment by cloudhan** [comment]
I am using the python binding for some testing purpose. Do we have the API or example?

**Comment by cloudhan** [comment]
I am using VLOGS_DEFAULT extensively. I tried to set session option and run options with `log_severity_level` and `log_verbosity_level` properly set. None of those affect verbosity level for default logger. I think we should at least provide a way to config it.

File: onnxruntime/python/onnxruntime_pybind_state.cc [hunk 1]
```diff
@@ -887,6 +887,12 @@ void addGlobalMethods(py::module& m, Environment& env) {
         default_logging_manager->SetDefaultLoggerSeverity(static_cast<logging::Severity>(severity));
       },
       "Sets the default logging severity. 0:Verbose, 1:Info, 2:Warning, 3:Error, 4:Fatal");
+  m.def(
+      "set_default_logger_verbosity", [&env](int vlog_level) {
+        logging::LoggingManager* default_logging_manager = env.GetLoggingManager();
+        default_logging_manager->SetDefaultLoggerVerbosity(vlog_level);
+      },
+      "Sets the default logging severity. 0:Verbose, 1:Info, 2:Warning, 3:Error, 4:Fatal");
   m.def(
```
Comment (skottmckay) [inline onnxruntime/python/onnxruntime_pybind_state.cc:896]: Please fix the documentation. Would be good to mention it requires severity to be verbose when doing so.

**Review by skottmckay — APPROVED** [review]

**Review by skottmckay — APPROVED** [review]

**Pull request merged**